controller/index.js

```diff
@@ -1,1019 +1,1054 @@
+   * @private
+   */
+  controller._returnProcessing = function (req, res, info) {
+    var table = controller._createTableKey('agol', req.params)
 
-        req.query.name = name;
-        // set the geometry type so the exporter can do its thing for csv points (add x,y)
-        if (itemJson.data && itemJson.data[0] && itemJson.data[0].info && itemJson.data[0].info.geometryType){
-          req.query.geomType = itemJson.data[0].info.geometryType;
+    if (typeof req.params.silent === 'undefined') {
+      agol.getCount(table, {}, function (err, count) {
+        if (err) {
+          return res.status(500).send(err)
         }
-        // force export of large data
-        req.query.large = true
-
-        agol.exportFile( fileParams, req.query, function(err, result){
-          if (result && result.status && result.status == 'processing'){
-            agol.getCount(table, {}, function(err, count){
-              var code = 202;
-              var response = {
-                status: 'processing',
-                processing_time: ( Date.now() - result.retrieved_at)/1000 || 0, 
-                count: count
-              };
-              if ( result.generating ){
-                response.generating = result.generating;
-                // we received an error from the server
-                if (result.generating.error){
-                  code = 502;
-                }
-              }
-              res.status(code).json( response );
-            });
-          } else if ( req.query.url_only ){
-            var origUrl = req.originalUrl.split('?');
-            origUrl[0] = origUrl[0].replace(/json/,req.params.format);
-            var newUrl = req.protocol +'://'+req.get('host') + origUrl[0] + '?' + origUrl[1].replace(/url_only=true&|url_only=true|/,'').replace('format='+req.params.format,'').replace('&format='+req.params.format,'');
-            res.json({url: newUrl});
-           } else {
-            if (err) {
-              res.status(err.code || 400).send( err );
-            } else {
-              if (req.params.format == 'json' || req.params.format == 'geojson'){
-                res.contentType('text');
-              } 
-              res.sendfile(result);
-            }
-          }
-        });
-      } else if (itemJson && itemJson.data && itemJson.data[0]) {
+        var code = 202
 
-        var opts = {
-          isFiltered: req.query.isFiltered, 
-          name: name, 
-          outSR: req.query.outSR
-        };
+        // we need some logic around handling long standing processing times
+        var processingTime = (Date.now() - info.retrieved_at) / 1000 || 0
 
-        if (itemJson.metadata) {
-          opts.metadata = itemJson.metadata;
+        var response = {
+          status: 'processing',
+          processing_time: processingTime,
+          count: count
+        }
+        if (info.generating) {
+          response.generating = info.generating
+          if (info.generating.error) {
```

Comment:
We need to flatten this response and have more stages than processing, but would be a breaking change.